src/migrate_dependencies.js

```diff
@@ -0,0 +1,23 @@
+async function migrate({
+  artifacts,
+  deployer,
+  network,
+}) {
+  if (network === "development" || network == "coverage") {
+    // deploy DevDependencies
+    const Dependencies = artifacts.require("./DevDependencies.sol")
+    await deployer.deploy(Dependencies)
+
+    // deploy libraries
+    const BiMap = artifacts.require("IdToAddressBiMap.sol")
```

Comment:
How are these deployed on rinkeby or mainnet?

Reply:
This is one of the next steps. I think the cleanest would be to export them into another project and migrate them there. I was looking into gno-utils, or any other place where we collect library contracts
@bh2smith  and @fleupold, Would you guys be fine with moving the contracts?

Reply:
How about https://github.com/gnosis/solidity-data-structures

Reply:
I have already submitted a PR with these Libraries moved into `solidity-data-structures`

https://github.com/gnosis/solidity-data-structures/pull/3

Reply:
I agree with the comments here.

However, just want to point out that, in DutchX this dependencies being handled in their own project also made it a bit overcomplicated to do changes. We had to be working in several projects, bumping versions just for simple things. So sometimes is finding a balance about "avoid replication" and "avoid too many dependencies".

For example, right now, if we use the WETH, I wouldn't depend on gnosis-utils, just for that.

Reply:
I would also like to point out that these data structures are unlikely to change. In fact, this Merkle contract hasn't been touched/changed since we were doing plasma research (~ August 2018).